crates/ra_hir_def/src/path.rs

```diff
@@ -297,68 +297,63 @@ impl From<Name> for Path {
 }
 
 pub mod known {
-    use hir_expand::name;
+    use hir_expand::name::N;
 
     use super::{Path, PathKind};
 
+    macro_rules! P {
+        ($start:ident $(:: $seg:ident)*) => { Path::from_simple_segments(PathKind::Abs, vec![N![$start], $(N![$seg],)*]) };
+    }
+
     pub fn std_iter_into_iterator() -> Path {
-        Path::from_simple_segments(
-            PathKind::Abs,
-            vec![name::STD, name::ITER, name::INTO_ITERATOR_TYPE],
-        )
+        P![std::iter::IntoIterator]
     }
 
     pub fn std_ops_try() -> Path {
-        Path::from_simple_segments(PathKind::Abs, vec![name::STD, name::OPS, name::TRY_TYPE])
+        P![std::ops::Try]
```

Comment:
Can we make P![std::ops::try] the public API? We probably still should check for a whitelist of paths though

Reply:
Hmm you're right we could just do that... how would you do the path whitelist though?

Reply:
By creating a macro rules branch for each allowed path? Or passing paths through an id macro, which is partially defined